### PR TITLE
BaSP-MR-100: Trainers resource adapted to Redux

### PR DESCRIPTION
**Pull request by MauroCaffesse**
A significant update was carried out on the "Trainers" resource of the project. Previously, this resource was developed using React as the main library, but it was decided to adapt it to Redux to take advantage of the benefits of global state management provided by this library.
The main difference that Redux introduces compared to React is the implementation of a centralized store that stores the application state globally. This means that data can be easily shared and accessed from any component, thus avoiding the need to explicitly pass data through multiple intermediate components.
By using Redux in the "Trainers" resource, a dedicated store has been established to store and manage the data related to the trainers. This implies that any component within the application can access this data without the need to pass it as props through multiple levels of component hierarchy. Additionally, Redux facilitates data updates and maintains a unidirectional flow, which improves predictability and the ability to track changes in the state, resulting in greater efficiency and scalability in data management.

### CODE REVIEW

File: src/Redux/Trainers/actions.js
Comment (francoax) [inline src/Redux/Trainers/actions.js:1]: The `SHOW_RESPONSE_MODAL` and `HIDE_RESPONSE_MODAL` are not necessary any more. You can delete them

File: src/Redux/Trainers/constants.js
Comment (francoax) [inline src/Redux/Trainers/constants.js:1]: Same here, both constants can be deleted

File: src/Redux/Trainers/reducers.js
Comment (francoax) [inline src/Redux/Trainers/reducers.js:1]: and here

**Review by francoax — CHANGES_REQUESTED** [review]

**Comment by francoax** [comment]
When you delete a trainer, the list from the table it's not updated and I still visualizing the trainer "deleted"

**Review by francoax — APPROVED** [review]

**Comment by francoax** [comment]
Nice work !!!!

File: src/Components/Trainers/Form/index.jsx [hunk 1]
```diff
@@ -160,67 +85,31 @@ const Form = () => {
     return true;
   };
 
+  const formFields = [
```
Comment (victoriassol) [inline src/Components/Trainers/Form/index.jsx:88]: Very cool what you did there!

**Review by victoriassol — APPROVED** [review]

File: src/Components/Trainers/Table/index.jsx [hunk 2]
```diff
@@ -1,26 +1,55 @@
...
-      </tbody>
-    </table>
+    <>
+      <div className={styles.filter}>
+        {filter.length === 0 ? <p>There is no trainer with that name or lastname</p> : ''}
+        <div className={`${styles['filter-container']}`}>
+          <input
+            className={`${styles['filter-input']}`}
+            type="text"
+            placeholder="Search by Name or Lastname"
```
Comment (victoriassol) [inline src/Components/Trainers/Table/index.jsx]:
```suggestion
            placeholder="Search by name or last name"
```

**Comment by gabiholl** [comment]
Good job! works great!

**Pull request merged**